modules/core/eks/stack.py

```diff
@@ -121,16 +135,21 @@ def __init__(
         cluster_admin_role.add_to_principal_policy(iam.PolicyStatement.from_json(cluster_admin_policy_statement_json_2))
         cluster_admin_role.add_to_principal_policy(iam.PolicyStatement.from_json(cluster_admin_policy_statement_json_3))
 
+        # KMS key for Envelope encryption
+        secrets_key = kms.Key(self, "SecretsKey")
```

Comment:
By adding this in and not making this a configurable item, you are forcing all deployed clusters to change...this need to be reviewed...